### PR TITLE
Add GraphQL sections to code quality

### PR DESCRIPTION
**Pull request by jakubcolony**

### CODE REVIEW

File: codeQuality/index.md [hunk 1]
```diff
@@ -6,6 +6,26 @@ In the CDapp we have linting rules enabled. When commiting, a number of pre-comm
 
 We also use Prettier to ensure that our code is formatted neatly on the page. Ensure you have the relevant editor extension installed.
 
+## GraphQL codegen
+
+CDapp has a codegen tool set up that automatically generates TS types based on the Amplify schema and client-side queries, mutations and subscriptions. It can be run with `npm run codegen` and produces a `generated.ts` file that contains the following: 
+
+### Amplify schema types
+
+Interfaces and enums for all types defined in the `schema.graphql` (and additional types added during Amplify schema compilation). Note those interfaces will contain ALL possible fields of each object type. Apart from enums, entire object types imported from `~gql` should almost never be used (use fragments instead).
```
Comment (rdig) [inline codeQuality/index.md:15]: Maybe link to amplify docs

File: codeQuality/index.md [hunk 2]
```diff
@@ -6,6 +6,26 @@ In the CDapp we have linting rules enabled. When commiting, a number of pre-comm
...
+
+### Client-side operations and fragments from `graphql/**/*.graphql`
+
+Interfaces for return types and variables, individual Apollo hooks for each operation as well as `Document` objects which can be useful when using Apollo client directly or refetching queries. Those return types will contain the exact subset of fields that the operations are fetching. If you expect to use the generated type of a particular operation or field, consider using fragments.
+
+Each fragment will also generate a corresponding interface. **Please make sure to re-export all fragments from `graphql/types.ts` to avoid confusion between fragments and entire object types from schema**
+
+## GraphQL best practices
```
Comment (rdig) [inline codeQuality/index.md:23]: Add a mention, or examples of how we do:
- polling
- pagination

Comment (rdig) [inline codeQuality/index.md:23]: Also mention subscriptions, but note that it's still "experimental" or something like that

**Review by rdig — APPROVED** [review]
Left some notes, but otherwise all good

File: codeQuality/index.md (hunk 2 above)
Comment (jakubcolony) [reply codeQuality/index.md:23]: I added a mention of `getSafePollingInterval` as good practice, I think the rest should live under architecture page: https://github.com/JoinColony/ColonyFrontEndLivingStandard/blob/master/architecture/index.md

File: codeQuality/index.md (hunk 1 above)
Comment (jakubcolony) [reply codeQuality/index.md:15]: 👍

**Review by rdig — APPROVED** [review]
Nicely done.

**Review by rumzledz — APPROVED** [review]
LGTM!

**Pull request merged**